public/app/core/components/TraceToLogs/KeyValueInput.tsx

```diff
@@ -0,0 +1,100 @@
+import { css } from '@emotion/css';
+import { GrafanaTheme, KeyValue } from '@grafana/data';
+import { SegmentInput, useStyles, InlineLabel, Icon } from '@grafana/ui';
+import React from 'react';
+
+const EQ_WIDTH = 3; // = 24px in inline label
+
+interface Props {
+  values: Array<KeyValue<string>>;
+  onChange: (values: Array<KeyValue<string>>) => void;
+  id?: string;
+  keyPlaceholder?: string;
+  valuePlaceholder?: string;
+}
+
+const KeyValueInput = ({
```

Comment:
Could this component be part of Grafana UI? Don't we have something similar (or planning to have) for new Query Editors?

Reply:
Brought it up and there are a few other in-progress features using similar inputs so we're going to coordinate making a more reusable component.